src/odemis/driver/picoquant.py

```diff
@@ -453,9 +695,9 @@ def SetInputCFD(self, channel, level, zc):
         level (int) CFD discriminator level in millivolts
         zc (0<=int): CFD zero cross in millivolts
         """
-        assert(channel in {0, 1})
-        assert(DISCRMIN <= level <= DISCRMAX)
-        assert(ZCMIN <= zc <= ZCMAX)
+        assert channel in {0, 1}
+        assert PH_DISCRMIN <= level <= PH_DISCRMAX
+        assert PH_ZCMIN <= zc <= PH_ZCMAX
```

Comment:
Shouldn't asserts like these have a logging message in case they fail? Just so that if a retry isn't succeding multiple times or some setting isn't within limits and therfore you eventually get into a timeout or other error you can find the trace/origin in the log.
There are quite some asserts in this PR but I don't see many logging messages (or custom error) accompanying them.

Reply:
I'm really not a fan of extra message in asserts. One of the main purpose of the asserts is to give some context to the reader. The short asserts allow to keep the line very simple and readable. The other purpose to stop early possible errors. On such case, the log will already indicate the line where it fails and all the backtrace. So for instance here you'd know that channel is not in {0,1}. Arguably, it'd be even nicer if we knew what the channel was, but I don't think the time and risk to make a mistake in the message is worthy. The chances that such assert trigger are very low.